src/vantage6_strongaya_general/general_statistics.py

```diff
@@ -0,0 +1,1415 @@
+    """
+    if inliers is None:
+        safe_log(
+            "warn",
+            "No inliers provided, returning all values as inliers and no outliers",
+        )
+        return column_values, pd.Series(dtype="Float64")
+
+    # Use explicit datatype if provided, otherwise fall back to dtype inspection
+    if datatype == "categorical" or (
+        datatype is None and isinstance(column_values.dtype, pd.CategoricalDtype)
+    ):
+        # Categorical variable - inliers is a list of allowed values
+        inliers_series = column_values[column_values.index.isin(inliers)]
+        outliers_series = column_values[~column_values.index.isin(inliers)]
+    elif datatype == "numerical" or (
+        datatype is None and pd.api.types.is_numeric_dtype(column_values)
+    ):
+        # Numerical variable - inliers should be a 2-element range [min, max]
+        if not isinstance(inliers, list) or len(inliers) != 2:
+            safe_log(
+                "warn",
+                f"For numerical variables, inliers must be a 2-element list [min, max]. Got: {inliers}. "
+                "Proceeding without determining outliers",
+            )
+            inliers_series = column_values
+            outliers_series = pd.Series(dtype="Float64")
+        elif not all(
+            isinstance(x, (int, float)) and type(x) is not bool for x in inliers
+        ):
+            safe_log(
+                "warn",
+                f"For numerical variables, inliers must contain numeric values. Got: {inliers}. "
+                "Proceeding without determining outliers",
+            )
+            inliers_series = column_values
+            outliers_series = pd.Series(dtype="Float64")
+        else:
+            inliers_series = column_values[
+                (column_values >= inliers[0]) & (column_values <= inliers[1])
+            ]
+            outliers_series = column_values[
+                (column_values < inliers[0]) | (column_values > inliers[1])
+            ]
+    else:
+        safe_log(
+            "warn",
+            "Expected datatype to be 'categorical' or 'numerical'. Proceeding without determining outliers",
+        )
+        inliers_series = column_values
+        outliers_series = pd.Series(dtype="Float64")
+
+    return inliers_series, outliers_series
+
+
+def _compute_local_mean(column_values: pd.Series) -> Union[int, float]:
+    """
+    Compute the mean using the local sum and non-NA rows.
+
+    Args:
+        column_values (pd.Series): Series of column values to compute the mean for.
+
+    Returns:
+        Union[int, float]: The local mean.
+    """
+    sum_val = safe_calculate(_compute_local_sum, 0.0, column_values=column_values)
+    rows = safe_calculate(
+        _compute_local_number_of_rows, 1, column_values=column_values, drop_na=True
+    )
+
+    # Avoid division by zero
+    return sum_val / rows if rows > 0 else 0.0
```

Comment:
If there are too few rows to share data, where are you preventing this data from sharing?